components/page-templates/B2B/CreateNewQuoteTemplate/CreateNewQuoteTemplate.tsx

```diff
@@ -0,0 +1,267 @@
+import React, { useState } from 'react'
+
+import ArrowBackIos from '@mui/icons-material/ArrowBackIos'
+import { LoadingButton } from '@mui/lab'
+import { Stack, Typography, Box, Grid, useMediaQuery, Theme, InputLabel } from '@mui/material'
+import { useTranslation } from 'next-i18next'
+
+import { createNewQuoteTemplateStyles } from './CreateNewQuoteTemplate.style'
+import { B2BProductDetailsTable, B2BProductSearch } from '@/components/b2b'
+import { CartItemList } from '@/components/cart'
+import { KiboTextBox } from '@/components/common'
+import { useAuthContext } from '@/context'
+import {
+  useGetPurchaseLocation,
+  useGetStoreLocations,
+  useProductCardActions,
+  useGetB2BUserQueries,
+  useDeleteQuoteItem,
+} from '@/hooks'
+import { useQuoteActions } from '@/hooks/custom/useQuoteActions/useQuoteActions'
+import { FulfillmentOptions as FulfillmentOptionsConstant } from '@/lib/constants'
+import { orderGetters, productGetters, quoteGetters } from '@/lib/getters'
+
+import { CrOrderItem, Location, Quote } from '@/lib/gql/types'
+
+export interface CreateNewQuoteTemplateProps {
+  quote: Quote
+  onAccountTitleClick: () => void
+}
+
+const CreateNewQuoteTemplate = (props: CreateNewQuoteTemplateProps) => {
+  const { quote, onAccountTitleClick } = props
+  const { t } = useTranslation('common')
+  const mdScreen = useMediaQuery((theme: Theme) => theme.breakpoints.up('md'))
+  const { user } = useAuthContext()
+
+  const { data } = useGetB2BUserQueries({
+    accountId: user?.id as number,
+    filter: `userId eq ${quote?.userId}`,
+  })
+  const quoteNumber = quoteGetters.getNumber(quote)
+  const quoteStatus = quoteGetters.getStatus(quote)
+  const createDate = quoteGetters.getQuoteCreateDate(quote)
+  const expirationDate = quoteGetters.getQuoteExpirationData(quote) ?? '-'
+  const accountName = user?.companyOrOrganization
+  const createdBy = (data?.items?.[0]?.firstName + ' ' + data?.items?.[0]?.lastName) as string
+  const quoteItems = quote?.items as CrOrderItem[]
+  const quoteId = quoteGetters.getQuoteId(quote)
+
```

Comment:
Instead of calling quoteGetters multiple times wouldn't it be nice to call it once?

Reply:
Resolved